### PR TITLE
Flask RESTX refactor

### PR DESCRIPTION
**Pull request by biegel**
This includes some of the Swagger docs as well.

### CODE REVIEW

**Comment by biegel** [comment]
Note: the BDD tests are failing because of authorization errors (HTTP 401). I need to add an auth token fields into the UI.  Will do that tomorrow!

File: tests/test_routes.py [hunk 1]
```diff
@@ -72,31 +77,39 @@ def _create_items(self, count):
             items.append(test_items)
         return items
 
+    #def test_bad_db_connection(self):
+    #    db.session.remove()
+    #    test_wishlist = WishlistFactory()
+    #    resp = self.app.post(
+    #        BASE_URL, json=test_wishlist.serialize(), content_type=CONTENT_TYPE_JSON, headers=self.headers
+    #    )
+    #    self.assertEqual(resp.status_code, status.HTTP_503_SERVICE_UNAVAILABLE)
```
Comment (billykern) [inline tests/test_routes.py]: remove this?

**Review by billykern — APPROVED** [review]
lgtm pending the auth changes / behave fixes

**Review by billykern — APPROVED** [review]
still lgtm

**Comment by biegel** [comment]
IBM Cloud [toolchain](https://cloud.ibm.com/devops/toolchains/c836ff1f-3847-4491-890e-32fa8a8c479d?env_id=ibm:yp:us-south "nyu-wishlist-service-sum21-toolchain"): [Delivery Pipeline](https://cloud.ibm.com/devops/pipelines/91c3adc5-dd63-41ae-ae8c-27f47731cb24 "nyu-wishlist-service-sum21") deployed [wishlists](https://github.com/NYU-DevOps-Squad-Wishlists/wishlists/releases/tag/deploy-ibm-yp-us-south-vbg5833-nyu.edu-dev-20210810-201500 "deploy-ibm-yp-us-south-vbg5833-nyu.edu-dev-20210810-201500") to [dev](https://console.bluemix.net/apps/157d30a4-e111-43a5-96d2-fd0b17f27b1a?env_id=ibm:yp:us-south "ibm:yp:us-south:vbg5833@nyu.edu:dev"), including commit 6b5e142b0ea0fe1cf00d72507a7eba2e18b30e5f

**Comment by biegel** [comment]
IBM Cloud [toolchain](https://cloud.ibm.com/devops/toolchains/c836ff1f-3847-4491-890e-32fa8a8c479d?env_id=ibm:yp:us-south "nyu-wishlist-service-sum21-toolchain"): [Delivery Pipeline](https://cloud.ibm.com/devops/pipelines/91c3adc5-dd63-41ae-ae8c-27f47731cb24 "nyu-wishlist-service-sum21") deployed [wishlists](https://github.com/NYU-DevOps-Squad-Wishlists/wishlists/releases/tag/deploy-ibm-yp-us-south-vbg5833-nyu.edu-prod-20210811-010226 "deploy-ibm-yp-us-south-vbg5833-nyu.edu-prod-20210811-010226") to [prod](https://console.bluemix.net/apps/3ab92089-100f-412a-adf2-90939fc24e98?env_id=ibm:yp:us-south "ibm:yp:us-south:vbg5833@nyu.edu:prod"), including commit 6b5e142b0ea0fe1cf00d72507a7eba2e18b30e5f

**Pull request merged**